.github/scripts/generate-wiki-page.js

```diff
@@ -0,0 +1,675 @@
+    }
+
+    if (ws.additional_files.metadata > 0) {
+      structureBadges.push('<span title="Metadata available">🟢</span>');
+    } else {
+      structureBadges.push('<span title="Metadata missing">🔴</span>');
+    }
+
+    if (ws.has_pending_prs && ws.pr_numbers.length > 0) {
+      const prNum = ws.pr_numbers[0];
+      const prUrl = `https://github.com/${repoName}/pull/${prNum}`;
+      const prTooltip = `Pending update PR #${prNum}`;
+      structureBadges.push(`[<span title="${prTooltip}">🔴</span>](${prUrl})`);
+    }
+
+    const structure = structureBadges.join('<br>');
+    const overlayRepoUrl = `https://github.com/${repoName}/tree/${branchName}/workspaces/${ws.name}`;
+    const workspaceName = `[${ws.name}](${overlayRepoUrl})`;
+
+    let source = 'N/A';
+    if (ws.repo_url && ws.commit_sha) {
+      const repoNameOnly = ws.repo_url.replace('https://github.com/', '');
+      if (ws.repo_flat) {
+        const sourceUrl = `${ws.repo_url}/tree/${ws.commit_sha}`;
+        source = `[${repoNameOnly}@${ws.commit_short}](${sourceUrl})`;
+      } else {
+        const workspacePath = `workspaces/${ws.name}`;
+        const sourceUrl = `${ws.repo_url}/tree/${ws.commit_sha}/${workspacePath}`;
+        source = `[${repoNameOnly}@${ws.commit_short}](${sourceUrl})`;
+      }
+    }
+
+    const overlayVersion = ws.overlay_backstage_version;
+    const sourceVersion = ws.source_backstage_version;
+    const displayVersion = sourceVersion || overlayVersion;
+
+    let backstageVersion = 'N/A';
+    if (displayVersion) {
+      if (overlayVersion && sourceVersion && overlayVersion !== sourceVersion) {
+        const tooltip = `Overlay overrides upstream version ${sourceVersion} to ${overlayVersion}`.replace(/"/g, '&quot;');
+        backstageVersion = `\`${displayVersion}\` <span title="${tooltip}">⚠️</span>`;
+      } else {
+        backstageVersion = `\`${displayVersion}\``;
+      }
+    }
+
+    let pluginsList = 'No plugins';
+    if (ws.plugins && ws.plugins.length > 0) {
+      const pluginsListItems = ws.plugins.map(p => {
+        const nameVer = p.details;
+        const status = p.status;
+
+        let icon, tooltip;
+        if (status === 'Supported') {
+          icon = '🟢';
+          tooltip = 'Red Hat Supported';
+        } else if (status === 'Community') {
+          icon = '🟡';
+          tooltip = 'Community / Dev Preview';
+        } else {
+          icon = '▪️';
+          tooltip = 'Community / Unknown';
+        }
+
+        return `<span title="${tooltip}">${icon}</span> <sub>\`${nameVer}\`</sub>`;
+      });
+
+      pluginsList = pluginsListItems.join('<br>');
+    }
+
+    md.push(`| ${structure} | ${workspaceName} | ${source} | ${backstageVersion} | ${pluginsList} |`);
+  }
+
+  md.push('');
+  md.push('---');
+  md.push('');
+
+  return md.join('\n');
+}
+
+async function main() {
+  const branchName = process.env.BRANCH_NAME || 'main';
```

Comment:
I guess you should use `core.getInput()` here, with environment variables starting with `INPUT_` in the calling yaml file.